plugins/inputs/nftables/nftables.go

```diff
@@ -0,0 +1,102 @@
+//go:generate ../../../tools/readme_config_includer/generator
+//go:build linux
+
+package nftables
+
+import (
+	_ "embed"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"os/exec"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/inputs"
+)
+
+//go:embed sample.conf
+var sampleConfig string
+
+var (
+	// defaultTables is the default nftables that we process
+	defaultTables = []string{"filter"}
+)
+
+type Nftables struct {
+	UseSudo bool     `toml:"use_sudo"`
+	Binary  string   `toml:"binary"`
+	Tables  []string `toml:"tables"`
+}
+
+func (*Nftables) SampleConfig() string {
+	return sampleConfig
+}
+
+func (nft *Nftables) Init() error {
+	if len(nft.Tables) == 0 {
+		return errors.New("invalid configuration - expected a 'Tables' entry with list of nftables to monitor")
+	}
+	return nil
+}
```

Comment:
How about

```golang
func (nft *Nftables) Init() error {
	if len(nft.Tables) == 0 {
		nft.Tables = []string{"filter"}
	}
	return nil
}
```

instead and removing the assignment in `init()`?

Altogether I would do

```suggestion
func (n *Nftables) Init() error {
	// Set defaults
	if len(n.Tables) == 0 {
		n.Tables = []string{"filter"}
	}
	if n.Binary == "" {
		n.Binary = "nft"
	}
	
	// Construct the command
	n.args := make([]string, 0, 3)
	if n.UseSudo {
		n.args = append(n.args, n.Binary)
		n.Binary = "sudo"
	}
	n.args = append(n.args, "--json", "list", "table")
	
	return nil
}
```